crypto/evp_extra/pq_sig_kat_tests_dilithium3.cc

```diff
@@ -0,0 +1,85 @@
+// Copyright Amazon.com, Inc. or its affiliates. All Rights Reserved.
+// SPDX-License-Identifier: Apache-2.0
+
+#include <gtest/gtest.h>
+#include <openssl/evp.h>
+#include "../rand_extra/pq_custom_randombytes.h"
+#include <vector>
+#include "../fipsmodule/evp/internal.h"
+#include "../dilithium/sig_dilithium.h"
+#include "../test/file_test.h"
+#include "../test/test_util.h"
+#include "./internal.h"
+
+ static void RunTest(FileTest *t)
+{
+  std::string count, mlen, smlen;
+  std::vector<uint8_t> seed, msg, pk, sk, sm;
+
+  ASSERT_TRUE(t->GetAttribute(&count, "count"));
+  ASSERT_TRUE(t->GetBytes(&seed, "seed"));
+  ASSERT_TRUE(t->GetAttribute(&mlen, "mlen"));
+  ASSERT_TRUE(t->GetBytes(&msg, "msg"));
+  ASSERT_TRUE(t->GetBytes(&pk, "pk"));
+  ASSERT_TRUE(t->GetBytes(&sk, "sk"));
+  ASSERT_TRUE(t->GetAttribute(&smlen, "smlen"));
+  ASSERT_TRUE(t->GetBytes(&sm, "sm"));
+
+  //the KAT files generated by the dilithium team use the optional APIs that create a
+  //signature for a message m and append the message to the end of the signature.
+  //We only want to bring the APIs that create and verify just the signature,
+  //therefore each signature is a constant DILITHIUM3_SIGNATURE_BYTES and we truncate
+  //the signed message down to a constant DILITHIUM3_SIGNATURE_BYTES.
+  uint8_t signature[DILITHIUM3_SIGNATURE_BYTES];
+  size_t signature_len = DILITHIUM3_SIGNATURE_BYTES;
+  sm.resize(signature_len);
+
+  //convert string read from KAT to int (is there a cleaner way?)
```

Comment:
You want `stoi()`.

Reply:
fixed in https://github.com/awslabs/aws-lc/pull/637/commits/d4aaf17d57b9852b74b3eee841e87e4ad2cbd339